backend/src/bundles/users/user.service.ts

```diff
@@ -47,9 +51,50 @@ class UserService implements Service {
     }
 
     public async update(
+        query: Record<string, number>,
+        payload: UserUpdateProfileRequestDto,
+    ): Promise<UserAuthResponseDto | null> {
+        try {
+            const userId = query['id'] as number;
+            const existingUser = await this.userRepository.find({
+                id: userId,
+            });
+            if (existingUser) {
+                const updatedUserDetails: Partial<UserDetailsModel> = {};
+                for (const property of Object.keys(payload)) {
+                    const value = payload[property];
+                    if (value) {
+                        updatedUserDetails[property] =
+                            property === 'weight' || property === 'height'
+                                ? Number(payload[property])
```

Comment:
Seems like an error will occur if users passes something like "70 kg" as weight, since type conversion will return NaN. 
Why do we store weight/height as numbers, but passing it in request as strings?

Reply:
In database and userDetailsModel weight/height are numbers, but in specifications for Profile Settings component it is written that user can input digits,letters etc. So, the type of weight/height in UserUpdateProfileRequestDto is string. So, I was extracting just numbers in a string on frontend and passing them to backend. And on backend convert string with numbers to a number type. 🫠

Reply:
So, users can enter any string, but we're only storing the numbers from that string, right? Do you find this weird? I would ask the authors of these specs about the reasoning behind this approach.

Reply:
So, these are specs)
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/72942456/43a22277-4351-4ad9-a6af-6ff685c7e23a)
And here is our database)
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/72942456/64bf2f95-ff5c-40c6-b296-e9fb3cea1fa1)